sim/warlock/rotations.go

```diff
@@ -414,7 +428,15 @@ func (warlock *Warlock) tryUseGCD(sim *core.Simulation) {
 				warlock.WaitUntil(sim, nextCD)
 				return
 			} else {
-				spell = filler
+				if warlock.Talents.Haunt &&
+					warlock.Haunt.CD.IsReady(sim) &&
+					rotationalLeeway[0]-fillerCastTime < 0 {
```

Comment:
Isn't there a possibility that there is a dps increase even if the filler finishes right after `rotationalLeeway[0] == 0`? We should check if that `< 0` could be changed to something positive although small and give a dps increase. (example check lines `		if nextBigCD - sim.CurrentTime > 0 && nextBigCD - sim.CurrentTime < fillerCastTime/10 {`).